src/test/java/teammates/test/cases/browsertests/InstructorCourseEditPageUiTest.java

```diff
@@ -160,6 +161,21 @@ private void testInviteInstructorAction() {
                 Const.StatusMessages.COURSE_REMINDER_SENT_TO + "InsCrsEdit.newInstr@gmail.tmt");
     }
 
+    private void testCancelAddingNewInstructor() {
+        ______TS("Cancelling the adding of a new instructor");
+        courseEditPage.clickShowNewInstructorFormButton();
+
+        ______TS("Click cancel but click no to confirmation prompt");
+        courseEditPage.clickDiscardChangesLink();
+        courseEditPage.waitForConfirmationModalAndClickCancel();
+        assertFalse(courseEditPage.verifyShowNewInstructorIsDisplayed());
+
+        ______TS("Click cancel but click yes to confirmation prompt");
+        courseEditPage.clickDiscardChangesLink();
+        courseEditPage.waitForConfirmationModalAndClickOk();
+        assertTrue(courseEditPage.verifyShowNewInstructorIsDisplayed());
```

Comment:
Not sure if I am understanding this test correctly.

You click `cancel`. Then modal comes up, and you click `ok`. Doesn't it mean that there should be no new instructor displayed instead because it's canceled?

Reply:
`courseEditPage.verifyShowNewInstructorIsDisplayed()` checks if the button `Add New Instructor` is displayed, which should be if the user clicks `'ok` as the instructor pane is gone.

The button `Add New Instructor` wouldn't be displayed after the user clicks on it.

Reply:
Ah. That's quite counter intuitive. Is there a way to test the presence of the form itself directly?

Reply:
Good point! I realized there is, through id `panelAddInstructor`. Will make that change :smile: